lib/PrintAsClang/_SwiftStdlibCxxOverlay.h

```diff
@@ -70,13 +70,14 @@ static_assert(sizeof(_impl::_impl_String) >= 0,
 SWIFT_INLINE_THUNK String::operator std::string() const {
   auto u = getUtf8();
   std::string result;
-  result.reserve(u.getCount() + 1);
```

Comment:
Are you sure that reserving the extra byte isn't necessary to avoid reallocation? I think it makes sense, since I believe the null terminator in `std::string` doesn't count as an element in the string, but I want to highlight it so we don't accidentally make a mistake here.

CC @hyp

Reply:
According to https://en.cppreference.com/w/cpp/string/basic_string/capacity.html:

> Note that the null terminator is not an element of the [std::basic_string](https://en.cppreference.com/w/cpp/string/basic_string.html)

So the behavior here looks correct to me.